app/controllers/users_controller.rb

```diff
@@ -0,0 +1,58 @@
+class UsersController < ApplicationController
+
+  before_action :authenticate_user!, only: [:edit_password]
+
+    def new
+      @user = User.new
+    end
+
+    def create
+      @user = User.new user_params
+
+      if @user.save
+        session[:user_id] = @user.id
+        redirect_to root_path, notice: "Your account has been created"
+      else
+        render:new
+      end
+    end
+
+    def edit
+      @user = User.find params[:id]
```

Comment:
Perhaps be cognizant of the DRY principle. User.find params[:id] could be refactored by making it into a private method called for example find_user and called using before_action.